frontend/src/components/organisms/EventRegisterForm.tsx

```diff
@@ -8,140 +8,172 @@ import Link from "next/link";
+	location: string;
+	datetime: string;
+	supervisors: string[];
+	capacity: number;
+	image_src: string;
+	tags: string[] | undefined;
+};
 
 interface EventRegisterFormProps {
-  eventid: string;
-  location: string;
-  datetime: string;
-  supervisors: string[];
-  capacity: number;
-  image_src: string;
-  tags?: string[];
+	eventDetails: eventData;
 }
 
+const url = BASE_URL as string;
+
 /**
  * A confirmation modal body
  */
 const ModalBody = ({
-  eventid,
-  handleClose,
+	eventid,
+	handleClose,
 }: {
-  eventid: string;
-  handleClose: () => void;
+	eventid: string;
+	handleClose: () => void;
 }) => {
-  const router = useRouter();
-  const register = () => {
-    router.replace(`/events/${eventid}/confirm/register`);
-  };
-  return (
-    <div>
-      <h2>Terms and Conditions</h2>
-      <p>
-        By registering, I agree that I will attend the event. If I cannot
-        attend, I will cancel my registration at least 24 hours before the event
-        begins. Failure to cancel my registration may negatively impact my
-        status as a volunteer
-      </p>
-      <Grid container spacing={2}>
-        <Grid item md={6} xs={12}>
-          <Button color="gray" type="button" onClick={handleClose}>
-            Disagree
-          </Button>
-        </Grid>
-        <Grid item md={6} xs={12}>
-          <Button color="dark-gray" type="button" onClick={register}>
-            Agree & Continue
-          </Button>
-        </Grid>
-      </Grid>
-    </div>
-  );
+	const router = useRouter();
+	const { user } = useAuth();
+
+	const register = async () => {
+		const userToken = await auth.currentUser?.getIdToken();
+
+		const attendeeid = await fetchUserIdFromDatabase(
+			user?.email as string,
+			userToken as string
+		);
+
+		const fetchUrl = `${url}/events/` + eventid + `/attendees`;
+		try {
+			const body = { attendeeid: attendeeid };
+			const response = await fetch(fetchUrl, {
+				method: "POST",
+				headers: {
+					Authorization: `Bearer ${userToken}`,
+					"Content-Type": "application/json",
+				},
+				body: JSON.stringify(body),
+			});
+			// Response Management
+			if (response.ok) {
+				const data = await response.json();
+			}
+		} catch (error) {
+		}
+
+		router.replace(`/events/${eventid}/register`);
+		window.location.reload(); // Call on reload to reroute page
+	};
+
+	return (
+		<div>
+			<h2>Terms and Conditions</h2>
+			<p>
+				By registering, I agree that I will attend the event. If I cannot
+				attend, I will cancel my registration at least 24 hours before the event
+				begins. Failure to cancel my registration may negatively impact my
+				status as a volunteer
+			</p>
+			<Grid container spacing={2}>
+				<Grid item md={6} xs={12}>
+					<Button color="gray" type="button" onClick={handleClose}>
+						Disagree
+					</Button>
+				</Grid>
+				<Grid item md={6} xs={12}>
+					<Button color="dark-gray" type="button" onClick={register}>
+						Agree & Continue
+					</Button>
+				</Grid>

```

Comment:
<img width = "500" src="https://github.com/cornellh4i/lagos-volunteers/assets/44273288/bf1c81a9-5151-4b18-880e-f2e9af04a254"/>

The `Disagree` and `Agree & Continue` should be grey and dark-grey but they're displaying on blue

Reply:
I think this is an issue with the Modal Inheriting the default theme set in _app.js. When we do Hifis and have a color for buttons, this will be fixed!